android/java/org/chromium/chrome/browser/settings/AppearancePreferences.java

```diff
@@ -281,7 +277,9 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                     .writeBoolean(BravePreferenceKeys.BRAVE_TAB_GROUPS_ENABLED, (boolean) newValue);
         } else if (PREF_BRAVE_ENABLE_SPEEDREADER.equals(key)) {
             UserPrefs.get(ProfileManager.getLastUsedRegularProfile())
-                    .setBoolean(BravePref.SPEEDREADER_PREF_ENABLED, (boolean) newValue);
+                    .setBoolean(
+                            BravePref.SPEEDREADER_PREF_ENABLED_FOR_ALL_SITES,
```

Comment:
Just working this out 😂 

This will trigger a restart when flipped... but only when `SPEEDREADER_PREF_ENABLED_FOR_ALL_SITES` changes and not `SPEEDREADER_PREF_FEATURE_ENABLED`.

That's only because `SPEEDREADER_PREF_FEATURE_ENABLED` is a new preference and isn't exposed to the user in the UI, right?

For the Origin case, should this also be considering `SPEEDREADER_PREF_FEATURE_ENABLED`? (for restart)

Reply:
This is actually not available on android unless you turn on a flag. We never actually shipped it on Android. So mainly I'm just keeping things the same there.